docs/050-breaking-changes.rst

```diff
@@ -0,0 +1,379 @@
+
+* The functions ``.call()``, ``.delegatecall()`, ``staticcall()``,
+  ``keccak256()``, ``sha256()`` and ``ripemd160()`` now accept only a single
+  ``bytes`` argument. Moreover, the argument is not padded. This was changed to
+  make more explicit and clear how the arguments are concatenated. Change every
+  ``.call()`` (and family) to a ``.call("")`` and every ``.call(signature, a,
+  b, c)`` to use ``.call(abi.encodeWithSignature(signature, a, b, c))`` (the
+  last one only works for value types).  Change every ``keccak256(a, b, c)`` to
+  ``keccak256(abi.encodePacked(a, b, c))``. Even though it is not a breaking
+  change, it is suggested that developers change
+  ``x.call(bytes4(keccak256("f(uint256)"), a, b)`` to
+  ``x.call(abi.encodeWithSignature("f(uint256)", a, b))``.
+
+* Functions ``.call()``, ``.delegatecall()`` and ``.staticcall()`` now return
+  ``(bool, bytes memory)``, since functions called with the opcode
+  ``STATICCALL`` might need to return data.  Change ``bool success =
+  otherContract.call("f")`` to ``(bool success, bytes memory data) =
+  otherContract.call("f")``.
+
+* Solidity now implements C99-style scoping rules for function local
+  variables, that is, variables can only be used after they have been
+  declared and only in the same or nested scopes. Variables declared in the
+  initialization block of a ``for`` loop are valid at any point inside the
+  loop.
+
+Explicitness Requirements
+=========================
+
+This section lists changes where the code now needs to be more explicit.
+
+* Explicit function visibility is now mandatory.  Add ``public`` to every
+  function and constructor, and ``external`` to every fallback or interface
+  function that does not specify its visibility already.
+
+* Explicit data location for all variables of struct, array or mapping types is
+  now mandatory. This is also applied to function parameters and return
+  variables.  For example, change ``uint[] x = m_x`` to ``uint[] storage x =
+  m_x``, and ``function f(uint[][] x)`` to ``function f(uint[][] memory x)``
+  where ``memory`` is the data location and might be replaced by ``storage`` or
+  ``calldata`` accordingly.  Note that ``external`` functions require
+  parameters with a data location of ``calldata``.
+
+* Variables of contract type do not contain ``address`` members anymore in
+  order to separate the namespaces.  Therefore, it is now necessary to
+  explicitly convert values of contract type to addresses before using an
+  ``address`` member.  Example: if ``c`` is a contract, change
+  ``c.transfer(...)`` to ``address(c).transfer(...)``.
+
+* The ``address`` type  was split into ``address`` and ``address payable``,
+  where only ``address payable`` provides the ``transfer`` function.  An
+  ``address payable`` can be directly converted to an ``address``, but the
+  other way around is not allowed. Converting ``address`` to ``address
+  payable`` is possible via conversion through ``uint160``.
```

Comment:
Perhaps explain that `address(c)` results in `address payable` if `c` has a payable callback function.